### PR TITLE
capsule & models refactor & users

### PR DESCRIPTION
**Pull request by jumperdevweb3**
1. Refactor models
2. applications/users login endpoints
3. Capsule integration
4. Vaults update
5. Middlewares update
6. API Token ---> AccessToken
7. Sessions update
8. Clean up & refactor

### CODE REVIEW

File: src/app/api/v1/applications/[appSlug]/tickets/deploy/route.tsx [hunk 1]
```diff
@@ -77,7 +77,7 @@ async function postHandler(req: NextRequestWithDeveloperUserAccessToken & NextRe
         name: contractName,
         developerId: req.developerId,
         version: nextId,
-        metadataUrl,
+        metadataImgUrl: metadataUrl,
```
Comment (adiusz) [inline src/app/api/v1/applications/[appSlug]/tickets/deploy/route.tsx:80]: nope, metadataUrl should stay the same.

Look at the return value from :
`const { metadataUrl, metadataImageUrl } = await uploadMetadata(`


iin the smartContractModel.create, you should go:

```
metadataUrl,
metadataImgUrl: metadataImageUrl
```

you can changed the name of the returned ImgUrl from `uploadMetadata` if you want it to match.

File: src/app/api/v1/applications/[appSlug]/users/login/route.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { NextRequest, NextResponse } from "next/server";
+import { StatusCodes } from "http-status-codes";
+import { sendVerificationEmailCode } from "@/lib/emails/auth/sendVerificationEmailCode";
+
+export async function POST(req: NextRequest) {
+  const body = await req.json();
+  const { email } = body;
+
+  const otpCodeSent = await sendVerificationEmailCode({
```
Comment (adiusz) [inline src/app/api/v1/applications/[appSlug]/users/login/route.ts:9]: here you will have a conflict cause i've splitted this function into 2 seperates one. 
1 for gerneating OTP, 2 for sending email (the generic one)

File: src/app/api/v1/applications/[appSlug]/users/verify/route.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+      const { data, error, status } = await refreshAccountSession(email, "user");
+      if (!!error) {
+        return NextResponse.json({ error }, { status });
+      }
+      return NextResponse.json(data, { status });
+    } else {
+      return NextResponse.json(
+        { error: "Invalid code", message: verifyEmailResult.message },
+        {
+          status: StatusCodes.BAD_REQUEST
```
Comment (adiusz) [inline src/app/api/v1/applications/[appSlug]/users/verify/route.ts:46]: reformat pls

File: src/app/api/v1/applications/[appSlug]/users/verify/route.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
+import { NextResponse } from "next/server";
+import { StatusCodes } from "http-status-codes";
+import { verifyEmailOtp } from "@/lib/emails/auth/verifyEmailOtp";
+import { userModel } from "@/models";
+import { refreshAccountSession } from "@/lib/auth/accounts/refreshAccountSession";
+import { getAppIdBySlug } from "@/lib/app";
+import { createUserAccount } from "@/lib/auth/accounts/createUserAccount";
+
+export async function POST(req: Request, { params: { appSlug } }) {
```
Comment (adiusz) [inline src/app/api/v1/applications/[appSlug]/users/verify/route.ts:9]: add validation pls. Use `src/app/api/v1/applications/[appSlug]/tickets/[id]/supply/route.ts` as the ref

File: src/app/api/v1/applications/[appSlug]/users/verify/route.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  const body = await req.json();
+  const { code } = body;
+
+  if (!code) {
+    return NextResponse.json({ error: "Invalid code format" }, {
+      status: StatusCodes.BAD_REQUEST
```
Comment (adiusz) [inline src/app/api/v1/applications/[appSlug]/users/verify/route.ts:15]: reformat pls

File: src/lib/1pwd-vault.ts [hunk 6]
```diff
@@ -3,125 +3,126 @@ import { shortenWalletAddress } from "@/utils/shortenWalletAddress";
...
           Authorization: `Bearer ${vaultToken}`,
-          "Content-Type": "application/json",
+          "Content-Type": "application/json"
         },
         body: JSON.stringify({
           vault: {
-            id: vaultId,
+            id: vaultId
           },
           // 🏗️ TODO: use title with real user's wallet/id
```
Comment (adiusz) [inline src/lib/1pwd-vault.ts:87]: this TODO is done if the title in 1pwd is filled correctly.

File: src/lib/1pwd-vault.ts [hunk 7]
```diff
@@ -171,24 +170,21 @@ export async function replaceVaultItemFields(
...
+  const vaultId = idPerType[type];
   try {
     const updatedItem = await fetch(
       `${vaultApiUrl}/v1/vaults/${vaultId}/items/${id}`,
       {
         method: "PATCH",
         headers: {
           Authorization: `Bearer ${vaultToken}`,
-          "Content-Type": "application/json",
+          "Content-Type": "application/json"
```
Comment (adiusz) [inline src/lib/1pwd-vault.ts:184]: youre using those headers in exact same form in 3 palces. pls unify if by making it a const

File: src/lib/auth/accounts/createCapsuleAccount.ts [hunk 8]
```diff
@@ -0,0 +1,24 @@
+import { capsule } from "@/lib/capsule";
+import { createVaultCapsuleKeyItem } from "@/lib/1pwd-vault";
+import { StatusCodes } from "http-status-codes";
+
+export const createCapsuleAccount = async (email: string, type: AccountType) => {
+  const hasWallet = await capsule.hasPregenWallet(email);
+  if (!hasWallet) {
+    try {
+      //@ts-ignore
```
Comment (adiusz) [inline src/lib/auth/accounts/createCapsuleAccount.ts:9]: no ts-ignore pls. just fix the error

File: src/lib/auth/sessions/updateDeveloperSession.ts [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+      refreshToken,
+      walletAddress: developer.walletAddress,
+      developerId: developer.id
+    };
+  } else {
+    const vaultItem = await createVaultAccessTokenItem("---", developer.id);
+    //wait for vault to be created
+    await new Promise((resolve) => setTimeout(resolve, 2000));
+    if (!vaultItem?.id) {
+      console.log("❌VAULT ITEM NOT CREATED");
```
Comment (adiusz) [inline src/lib/auth/sessions/updateDeveloperSession.ts:57]: remove console.logs pls

File: src/lib/auth/sessions/updateUserSession.ts [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+      data: {
+        userId: user.id,
+        appId,
+        accessToken,
+        refreshToken,
+        expiresAt: new Date(Date.now() + 2 * 24 * 60 * 60 * 1000)
+      }
+    });
+
+    console.log(`New session id: ${newSession.id}`);
```
Comment (adiusz) [inline src/lib/auth/sessions/updateUserSession.ts:58]: is it neccesary?

File: src/lib/emails/auth/validateEmail.ts [hunk 11]
```diff
@@ -1,23 +1,22 @@
 "use server";
```
Comment (adiusz) [inline src/lib/emails/auth/validateEmail.ts:1]: you will have the name conflict. pls use my name as its more descriptive

**Review by adiusz — CHANGES_REQUESTED** [review]

File: src/lib/auth/accounts/createCapsuleAccount.ts (hunk 8 above)
Comment (jumperdevweb3) [reply src/lib/auth/accounts/createCapsuleAccount.ts:9]: It's not on our side.

**Pull request merged**